[GStreamer] Don't advertise support for HDR on systems that don't explicitly declare that support

eocanha opened the pull request:
#### 989ce9fac92b416642852a72866eddd9fa0e55eb
<pre>
[GStreamer] Don&apos;t advertise support for HDR on systems that don&apos;t explicitly declare that support
<a href="https://bugs.webkit.org/show_bug.cgi?id=299873">https://bugs.webkit.org/show_bug.cgi?id=299873</a>

Reviewed by Philippe Normand.

Some websites and JS players can try to play HDR video on systems that
don&apos;t support it. Even when there isn&apos;t an effective HDR connected
screen detection mechanism yet in place on GStreamer ports, it would be
nice to at least report HDR as unsupported to those webpages asking for
it.

See: <a href="https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1564">https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1564</a>

This patch now filters the MediaCapabilities query, so if the webpage
is asking for an hdrMetadataType or a transferFunction and our
internal screen data doesn&apos;t acknowledge the presence of an HDR screen,
we reply that such MediaCapabilities combination is not supported.

Also, if the requested codecs aren&apos;t supported (even by software),
we now return that the MediaCapabilities combination is not supported.

Original author: Andrzej Surdej &lt;Andrzej_Surdej@comcast.com&gt;

* Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp:
(WebCore::GStreamerRegistryScanner::isConfigurationSupported const): Apply filters. Check if codecs are supported.

Canonical link: <a href="https://commits.webkit.org/301049@main">https://commits.webkit.org/301049@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d34484c5bd362123af0b75f6b12ab74e773aa206

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124393 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44073 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/34803 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131232 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76423 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/126270 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/44819 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/52672 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/94625 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/62775 "") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127347 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/35723 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111258 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75211 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/34664 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/29420 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/74709 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105471 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/29640 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/133897 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51291 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39122 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103104 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/51690 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107476 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/102899 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26276 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48267 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26516 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48228 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51158 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/56950 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/50581 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/53941 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52256 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

philn commented:
```
/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp:1030:36: error: ‘primaryScreenDisplayID’ was not declared in this scope
 1030 |         auto* scrData = screenData(primaryScreenDisplayID());
      |                                    ^~~~~~~~~~~~~~~~~~~~~~
/home/buildbot-worker/WPE-Build-EWS/build/Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp:1030:25: error: ‘screenData’ was not declared in this scope; did you mean ‘scrData’?
 1030 |         auto* scrData = screenData(primaryScreenDisplayID());
      |                         ^~~~~~~~~~
      |                         scrData
```

eocanha commented:
> error: ‘primaryScreenDisplayID’ was not declared in this scope

Sorry, this is embarrassing... It turns out that using WebKitGTK to check if a code ifdeffed for WPE works or even builds doesn't look like a great idea on my side. :facepalm:

I'm submitting a new version with the extra required includes.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d6c7252b109bf6b4fa0c359a83986e952cc0f394. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51577

philn commented on Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp:
```diff
@@ -1027,11 +1031,31 @@ GStreamerRegistryScanner::RegistryLookupResult GStreamerRegistryScanner::isConfi
             videoConfiguration.bitrate, videoConfiguration.framerate);
 #endif
 
+#if PLATFORM(WPE)
```
Can you remove this please?

eocanha committed:
[GStreamer] Don't advertise support for HDR on systems that don't explicitly declare that support
https://bugs.webkit.org/show_bug.cgi?id=299873

Reviewed by Philippe Normand.

Some websites and JS players can try to play HDR video on systems that
don't support it. Even when there isn't an effective HDR connected
screen detection mechanism yet in place on GStreamer ports, it would be
nice to at least report HDR as unsupported to those webpages asking for
it.

See: https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1564

This patch now filters the MediaCapabilities query, so if the webpage
is asking for an hdrMetadataType or a transferFunction and our
internal screen data doesn't acknowledge the presence of an HDR screen,
we reply that such MediaCapabilities combination is not supported.

Also, if the requested codecs aren't supported (even by software),
we now return that the MediaCapabilities combination is not supported.

Original author: Andrzej Surdej <Andrzej_Surdej@comcast.com>

* Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp:
(WebCore::GStreamerRegistryScanner::isConfigurationSupported const): Apply filters. Check if codecs are supported.

Canonical link: https://commits.webkit.org/301049@main
